src/types/state.d.ts

```diff
@@ -151,17 +176,86 @@ type LabelUISettings = {
   name: boolean;
   short: boolean;
 };
-
-type DownloadState = Record<string, DownloadProgress>;
+type CacheSettings = {
+  enabled: boolean;
+};
+type DownloadState = Record<string, DownloadProgress | CompletedDownload>;
 
 type DownloadProgress = {
   progress: number;
-  successful: null | boolean;
+  successful?: null | boolean;
   blobURL: string;
   size: null | number;
   version: number;
+  fileType: DownloadFileTypes;
+  date?: Date;
+  length?: number;
+  subset?: boolean;
 };
+
+type CompletedDownload = DownloadProgress & {
+  blob: Blob;
+  date: Date;
+  length: number;
+  subset: boolean;
+};
+
+type Server =
+  | 'api'
+  | 'ebi'
+  | 'ipScan'
+  | 'genome3d'
+  | 'wikipedia'
+  | 'alphafold'
+  | 'repeatsDB'
+  | 'disprot'
+  | 'proteinsAPI';
 type ServerStatus = {
-  status: boolean;
-  lastCheck: number;
+  status: boolean | null;
+  lastCheck: number | null;
+};
+
+type DownloadFileTypes =
+  | 'txt'
+  | 'accession'
+  | 'fasta'
+  | 'json'
+  | 'ndjson'
+  | 'tsv'
+  | 'xml';
+
+type DataProgress = Record<string, DatumProgress>;
+type DatumProgress = {
+  progress: number;
+  weight: number;
+};
+
+type FavouritesState = {
+  entries: Array<string>;
+};
+
+type ToastsState = Record<string, ToastData>;
+
+type ToastData = {
+  paused?: boolean;
+  className?: string;
+  title: string;
+  body?: string;
+  link?: { to: InterProPartialLocation; children: string };
+  action?: {
+    text: string;
+    fn: () => void;
+  };
+  ttl?: number;
+  checkBox?: {
+    label: string;
+    fn: () => void;
+  };
+  // handleClose?: () => void;
```

Comment:
So, none of the components using the Toast as an action injects the `handleClose` but the `ToastDisplay` which ends up acting like its manager, defines the prop.

Note to @apolignano: When migrating the `Toast` and `ToastDisplay`, it will remember to extend this object with the `handleClose` extra prop